crates/astria-composer/src/executor/mod.rs

```diff
@@ -226,10 +231,19 @@ impl Executor {
 
         let reset_time = || Instant::now() + self.block_time;
 
-        loop {
+        let reason = loop {
             select! {
                 biased;
 
+                () = self.shutdown_token.cancelled() => {
+                    info!("executor shutting down.");
+                    // do not accept any new txs during shutdown.
+                    self.serialized_rollup_transactions.close();
```

Comment:
Is this required?  Looks like the `bundle_factory` won't receive any more txs if we've exited this loop.

Reply:
yeah if we are shutting the executor down, we do not want to receive any more txs from the collectors.

Reply:
I understand the logic behind using `Receiver::close`, but I believe the channel is not currently drained after the loop.

Reply:
its a good point. the receve future might not have been polled yet and we would loose txs sent. I did not think of that. I will drain the channel!